### PR TITLE
chore(#?): Patch release v4.1.2 - Cherrypick #8173 and #8205

### PR DESCRIPTION
**Pull request by kennsippell**
# Description

Prepares branch for cht-core v4.1.2 patch release.

Cherry-picks #8173 and #8205. Minor conflicts were resolved easily. Conflicts in package-lock.json resolved by re-running `npm i` and committing the result.

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs

If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:4.1.2-8173-8205/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:4.1.2-8173-8205/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:4.1.2-8173-8205/docker-compose/cht-couchdb-clustered.yml)

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: api/package-lock.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "medic-api",
   "version": "0.1.0",
-  "lockfileVersion": 2,
+  "lockfileVersion": 3,
```
[the remark this answers is not in the document]
Comment (kennsippell) [reply api/package-lock.json]: Oops. [lockfileVersion 2](https://docs.npmjs.com/cli/v9/configuring-npm/package-lock-json#lockfileversion) is from npm 7 or 8 and I used 9.

I've regenerated with the minimum npm version in the package.json which is npm 8.3.1.

[the remark this answers is not in the document]
Comment (dianabarsan) [reply api/package-lock.json]: I would just cherry pick the changes again, without regenerating package locks. does that not work?

File: api/package-lock.json [hunk 2]
```diff
@@ -45,7 +45,9 @@
         "morgan": "^1.10.0",
         "mustache": "^4.2.0",
         "node-cache": "^5.1.2",
+        "node-fetch": "^2.6.9",
         "node-html-parser": "^3.0.4",
+        "npm": "^8.3.1",
```
Comment (garethbowen) [inline api/package-lock.json:50]: Something has gone horribly wrong with this cherry-pick  :/

**Comment by garethbowen** [comment]
I've cherry-picked directly into the `4.1.x` branch so this PR is no longer needed.

**Pull request closed without merging**